src/main/java/io/swagger/codegen/languages/kotlin/KotlinServerCodegen.java

```diff
@@ -0,0 +1,209 @@
+package io.swagger.codegen.languages.kotlin;
+
+import com.google.common.collect.ImmutableMap;
+import io.swagger.codegen.CliOption;
+import io.swagger.codegen.CodegenConstants;
+import io.swagger.codegen.CodegenType;
+import io.swagger.codegen.SupportingFile;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.File;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import com.github.jknack.handlebars.helper.StringHelpers;
+import com.github.jknack.handlebars.Handlebars;
+import org.apache.commons.lang3.StringUtils;
+
+public class KotlinServerCodegen extends AbstractKotlinCodegen {
+
+    public static final String DEFAULT_LIBRARY = Constants.KTOR;
+    static Logger LOGGER = LoggerFactory.getLogger(KotlinServerCodegen.class);
+    private Boolean autoHeadFeatureEnabled = true;
+    private Boolean conditionalHeadersFeatureEnabled = false;
+    private Boolean hstsFeatureEnabled = true;
+    private Boolean corsFeatureEnabled = false;
+    private Boolean compressionFeatureEnabled = true;
+
+    // This is here to potentially warn the user when an option is not supoprted by the target framework.
+    private Map<String, List<String>> optionsSupportedPerFramework = new ImmutableMap.Builder<String, List<String>>()
+            .put(Constants.KTOR, Arrays.asList(
```

Comment:
Please use `Collections.singletonMap(Constants.KTOR, Arrays.asList(..));` instead of guava.